zadania/Program.cs

```diff
@@ -101,15 +91,50 @@ void summary(List<Tuple<int,int,int,int>> links)
             summary[travel] = 1;
     }
     foreach (KeyValuePair<string,int> kvp in summary)
-        Console.WriteLine("Podróż {0} odbyła się {1} razy", kvp.Key, kvp.Value);
+        Console.WriteLine($"Podróż {kvp.Key} odbyła się {kvp.Value} razy");
+}
+void maxTo(List<Tuple<int,int,int,int>> links)
+{
+    Dictionary<Tuple<int,int>,int> amountPerHub = new Dictionary<Tuple<int, int>, int>();
+    for (int i = 0; i < links.Count; i++)
+    {
+        Tuple<int, int> key = new Tuple<int, int>(links[i].Item3, links[i].Item4);
+        if (amountPerHub.ContainsKey(key)) amountPerHub[key]++;
+        else amountPerHub.Add(key, 1);
+    }
+    var result = amountPerHub.MaxBy(x => x.Value);
+    log($"Najwięcej osób - {result.Value} - podróżowało do {result.Key.Item1},{result.Key.Item2}", false);
 }
-double[,] densityTable = generateTableDensity(width, height);
-double[,] desirabilityTable = generateTableDesirability(width, height);
+void maxFrom(List<Tuple<int,int,int,int>> links)
+{
+    Dictionary<Tuple<int, int>, int> amountPerHub = new Dictionary<Tuple<int, int>, int>();
+    for (int i = 0; i < links.Count; i++)
+    {
+        Tuple<int, int> key = new Tuple<int, int>(links[i].Item1, links[i].Item2);
+        if (amountPerHub.ContainsKey(key)) amountPerHub[key]++;
+        else amountPerHub.Add(key, 1);
+    }
+    var result = amountPerHub.MaxBy(x => x.Value);
+    log($"Najwięcej osób - {result.Value} - podróżowało z {result.Key.Item1},{result.Key.Item2}", true);
+}
```

Comment:
Spójrz na kod funkcji maxTo, nie wydaje ci się, że jest prawie identyczny? Wymyśl jak to scalić, ale pierwsze użyj GroupBy aby uprościć całego for-a